### DISCUSSION TITLE
PubSub Limitations?

### DISCUSSION THREAD

**Discussion by jeffery30162** [uNetworking/uWebSockets.js#683] [Q&A] [answered]
I am wondering the limitations of the pub/sub. Is there a limit to the number of subs? Lets say I am making a chat application and there are 1mill chat rooms/channels can I have a pub sub for each chat room?

**Comment by e3dio** [comment, marked as answer]
Only limited by your available RAM, CPU, and network capacity. A "topic" is just a string name and set of websockets subscribed to it. If there are no subscribers left the topic will remove itself so it won't take up any resources

**Comment by ghost** [comment]
Pub/sub in v20 is very lightweight and simple. You can have as many rooms as you want
